diff --git a/configure.ac b/configure.ac
index ba5bbf2..b512713 100644
--- a/configure.ac
+++ b/configure.ac
@@ -4,7 +4,6 @@ AC_DEFINE(API_VERSION, 9, api version)
 
 AM_INIT_AUTOMAKE
 AC_CONFIG_HEADERS(config.h)
-AM_PATH_CPPUNIT(1.9.6)
 
 AC_PROG_CXX
 AC_PROG_LIBTOOL
@@ -37,17 +36,20 @@ if test "x$ax_cv_ncursesw" != xyes && test "x$ax_cv_ncurses" != xyes; then
    AC_MSG_ERROR([requires either NcursesW or Ncurses library])
 fi
 
-CFLAGS="$CFLAGS $PTHREAD_CFLAGS $CURSES_CFLAGS"
-CXXFLAGS="$CXXFLAGS $PTHREAD_CFLAGS $CURSES_CFLAGS"
-LIBS="$PTHREAD_LIBS $CURSES_LIB $LIBS"
+PKG_CHECK_MODULES([CPPUNIT], [cppunit],, [no_cppunit="yes"])
+
+CFLAGS="$PTHREAD_CFLAGS $CPPUNIT_CFLAGS $CFLAGS $CURSES_CFLAGS"
+CXXFLAGS="$PTHREAD_CFLAGS $CPPUNIT_CFLAGS $CXXFLAGS $CURSES_CFLAGS"
+LIBS="$PTHREAD_LIBS $CURSES_LIB $CPPUNIT_LIBS $LIBS"
+
 
 PKG_CHECK_MODULES([libcurl], libcurl >= 7.15.4,
-	          CXXFLAGS="$CXXFLAGS $libcurl_CFLAGS";
-		  LIBS="$LIBS $libcurl_LIBS")
+                  CXXFLAGS="$CXXFLAGS $libcurl_CFLAGS";
+                  LIBS="$LIBS $libcurl_LIBS")
 
 PKG_CHECK_MODULES([libtorrent], libtorrent >= 0.13.6,
-	          CXXFLAGS="$CXXFLAGS $libtorrent_CFLAGS";
-		  LIBS="$LIBS $libtorrent_LIBS")
+                  CXXFLAGS="$CXXFLAGS $libtorrent_CFLAGS";
+                  LIBS="$LIBS $libtorrent_LIBS")
 
 AC_LANG_PUSH(C++)
 TORRENT_WITH_XMLRPC_C
@@ -61,19 +63,19 @@ TORRENT_CHECK_CACHELINE()
 TORRENT_CHECK_POPCOUNT()
 
 CC_ATTRIBUTE_UNUSED(
-	AC_DEFINE([__UNUSED], [__attribute__((unused))], [Wrapper around unused attribute]),
-	AC_DEFINE([__UNUSED], [], [Null-wrapper if unused attribute is unsupported])
+        AC_DEFINE([__UNUSED], [__attribute__((unused))], [Wrapper around unused attribute]),
+        AC_DEFINE([__UNUSED], [], [Null-wrapper if unused attribute is unsupported])
 )
 
 AC_OUTPUT([
-	Makefile
-	doc/Makefile
-	src/Makefile
-	src/core/Makefile
-	src/display/Makefile
-	src/input/Makefile
-	src/rpc/Makefile
-	src/ui/Makefile
-	src/utils/Makefile
+        Makefile
+        doc/Makefile
+        src/Makefile
+        src/core/Makefile
+        src/display/Makefile
+        src/input/Makefile
+        src/rpc/Makefile
+        src/ui/Makefile
+        src/utils/Makefile
         test/Makefile
 ])
